src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2436,7 +2447,168 @@ public void UpdateIsNullableEnabledForConstructorsAndFields(bool useStatic, bool
+            public readonly RecordDeclarationSyntax? RecordDeclarationWithParameters;
+            public readonly int StaticInitializersSyntaxLength;
+
+            /// <summary>
+            /// Index into <see cref="InstanceInitializers"/> for the set of initializers that we'll need
+            /// to patch when adding the initializers for record positional members.
+            /// </summary>
+            public readonly int InstanceInitializersIndexForRecordDeclarationWithParameters;
+            public readonly bool IsNullableEnabledForInstanceConstructorsAndFields;
+            public readonly bool IsNullableEnabledForStaticConstructorsAndFields;
+
+            public DeclaredMembersAndInitializers(
+                ImmutableArray<Symbol> nonTypeNonIndexerMembers,
+                ImmutableArray<ImmutableArray<FieldOrPropertyInitializer>> staticInitializers,
+                ImmutableArray<ImmutableArray<FieldOrPropertyInitializer.Builder>> instanceInitializers,
+                ImmutableArray<SyntaxReference> indexerDeclarations,
+                int staticInitializersSyntaxLength,
+                RecordDeclarationSyntax? recordDeclarationWithParameters,
+                int instanceInitializersIndexForRecordDeclarationWithParameters,
+                bool isNullableEnabledForInstanceConstructorsAndFields,
+                bool isNullableEnabledForStaticConstructorsAndFields)
+            {
+                Debug.Assert(!nonTypeNonIndexerMembers.IsDefault);
+                Debug.Assert(!staticInitializers.IsDefault);
+                Debug.Assert(!instanceInitializers.IsDefault);
+                Debug.Assert(!indexerDeclarations.IsDefault);
+
+                Debug.Assert(!nonTypeNonIndexerMembers.Any(s => s is TypeSymbol));
+                Debug.Assert(!nonTypeNonIndexerMembers.Any(s => s.IsIndexer()));
+                Debug.Assert(!nonTypeNonIndexerMembers.Any(s => s.IsAccessor() && ((MethodSymbol)s).AssociatedSymbol.IsIndexer()));
+
+                this.NonTypeNonIndexerMembers = nonTypeNonIndexerMembers;
+                this.StaticInitializers = staticInitializers;
+                this.InstanceInitializers = instanceInitializers;
+                this.IndexerDeclarations = indexerDeclarations;
+                this.StaticInitializersSyntaxLength = staticInitializersSyntaxLength;
+                this.RecordDeclarationWithParameters = recordDeclarationWithParameters;
+                this.InstanceInitializersIndexForRecordDeclarationWithParameters = instanceInitializersIndexForRecordDeclarationWithParameters;
+                this.IsNullableEnabledForInstanceConstructorsAndFields = isNullableEnabledForInstanceConstructorsAndFields;
+                this.IsNullableEnabledForStaticConstructorsAndFields = isNullableEnabledForStaticConstructorsAndFields;
+            }
+        }
+
+        private sealed class MembersAndInitializersBuilder
+        {
+            public ArrayBuilder<Symbol> NonTypeNonIndexerMembers { get; set; }
```

Comment:
>public ArrayBuilder<Symbol> NonTypeNonIndexerMembers { get; set; } [](start = 12, length = 66)

I am not sure why this is not a readonly field.

Reply:
`AddSynthesizedRecordMembersIfNecessary` can replace the array builder.

---
In reply to: [562871158](https://github.com/dotnet/roslyn/pull/50650#discussion_r562871158) [](ancestors = 562871158)